src/lib/statistics/statistics_objects/equal_distinct_count_histogram.cpp

```diff
@@ -0,0 +1,220 @@
+      std::vector<std::pair<T, HistogramCountType>>{value_distribution_map.begin(), value_distribution_map.end()};
+  std::sort(value_distribution.begin(), value_distribution.end(),
+            [&](const auto& l, const auto& r) { return l.first < r.first; });
+
+  return value_distribution;
+}
+}  // namespace
+
+namespace opossum {
+
+template <typename T>
+EqualDistinctCountHistogram<T>::EqualDistinctCountHistogram(std::vector<T>&& bin_minima, std::vector<T>&& bin_maxima,
+                                                            std::vector<HistogramCountType>&& bin_heights,
+                                                            const HistogramCountType distinct_count_per_bin,
+                                                            const BinID bin_count_with_extra_value,
+                                                            const HistogramDomain<T>& domain)
+    : AbstractHistogram<T>(domain),
+      _bin_minima(std::move(bin_minima)),
+      _bin_maxima(std::move(bin_maxima)),
+      _bin_heights(std::move(bin_heights)),
+      _distinct_count_per_bin(distinct_count_per_bin),
+      _bin_count_with_extra_value(bin_count_with_extra_value) {
+  Assert(_bin_minima.size() == _bin_maxima.size(), "Must have the same number of lower as upper bin edges.");
+  Assert(_bin_minima.size() == _bin_heights.size(), "Must have the same number of edges and heights.");
+  Assert(_distinct_count_per_bin > 0, "Cannot have bins with no distinct values.");
+  Assert(_bin_count_with_extra_value < _bin_minima.size(), "Cannot have more bins with extra value than bins.");
+
+  AbstractHistogram<T>::_assert_bin_validity();
+
+  _total_count = std::accumulate(_bin_heights.cbegin(), _bin_heights.cend(), HistogramCountType{0});
+  _total_distinct_count =
+      static_cast<HistogramCountType>(_distinct_count_per_bin * bin_count() + _bin_count_with_extra_value);
+}
+
+template <typename T>
+std::shared_ptr<EqualDistinctCountHistogram<T>> EqualDistinctCountHistogram<T>::from_column(
+    const Table& table, const ColumnID column_id, const BinID max_bin_count, const HistogramDomain<T>& domain) {
+  Assert(max_bin_count > 0, "max_bin_count must be greater than zero ");
+
+  auto value_distribution = value_distribution_from_column(table, column_id, domain);
+
+  if (value_distribution.empty()) {
+    return nullptr;
+  }
+
+  // If there are fewer distinct values than the number of desired bins use that instead.
+  const auto bin_count =
+      value_distribution.size() < max_bin_count ? static_cast<BinID>(value_distribution.size()) : max_bin_count;
+
+  // Split values evenly among bins.
+  const auto distinct_count_per_bin = static_cast<size_t>(value_distribution.size() / bin_count);
+  const BinID bin_count_with_extra_value = value_distribution.size() % bin_count;
+
+  std::vector<T> bin_minima(bin_count);
+  std::vector<T> bin_maxima(bin_count);
+  std::vector<HistogramCountType> bin_heights(bin_count);
+
+  auto current_bin_begin_index = BinID{0};
```

Comment:
Add a comment that `_index` means index into the list of distinct values? That took me some time to figure out.

Reply:
How about 

```c++
  // `bin_minimum_distinct_value_idx` and `bin_maximum_distinct_value_idx` are indices into `value_distribution`
  // describing which range of distinct values goes into a bin
  auto bin_minimum_distinct_value_idx = BinID{0};
  for (BinID bin_index = 0; bin_index < bin_count; bin_index++) {
    auto bin_maximum_distinct_value_idx = bin_minimum_distinct_value_idx + distinct_count_per_bin - 1;
```

Reply:
Maybe just `min_value_idx`? Also, is `BinID` the correct type here?